prbt_hardware_support/include/prbt_hardware_support/modbus_msg_in_utils.h

```diff
@@ -123,11 +125,20 @@ class ModbusMsgInBuilder
 
       msg->holding_registers.layout.data_offset = first_index_to_read;
       msg->holding_registers.layout.dim.push_back(std_msgs::MultiArrayDimension());
-      msg->holding_registers.layout.dim[0].size = tab_reg.size();
+      try
+      {
+        msg->holding_registers.layout.dim[0].size
+                                      = boost::numeric_cast<std_msgs::MultiArrayDimension::_size_type>(tab_reg.size());
+        msg->holding_registers.data = tab_reg;
+      }
+      catch(boost::numeric::positive_overflow& e)
+      {
+        ROS_ERROR("The modbus register is to big!");
```

Comment:
should this also fail?

Reply:
throw ModbusExceptionDisconnect ?

Reply:
I'd suggest:
1. **use a `static_cast<>`**. Since you currently do not use the overflow but simply proceed, you could as good use the standard cast.
2. **Not catch** (aborting the program gives the best possibilities to debug and most likely will issue bug reports, if someone breaks this part of the code in the future while ROS_ERRORs are silently ignored by most users